manual/io/index.md

```diff
@@ -56,784 +46,433 @@ root [1] TFile f1("my.root");
-TKey Name = h4,     Title = histo    nr:4,     Cycle = 1
-TKey Name = h5,     Title = histo    nr:5,     Cycle = 1
-TKey Name = h6,     Title = histo    nr:6,     Cycle = 1
-TKey Name = h7,     Title = histo    nr:7,     Cycle = 1
-TKey Name = h8,     Title = histo    nr:8,     Cycle = 1
-TKey Name = h9,     Title = histo    nr:9,     Cycle = 1
-TKey Name = h10,    Title = histo    nr:10,    Cycle = 1
-TKey Name = h11,    Title = histo    nr:11,    Cycle = 1
-TKey Name = h12,    Title = histo    nr:12,    Cycle = 1
-TKey Name = h13,    Title = histo    nr:13,    Cycle = 1
-TKey Name = h14,    Title = histo    nr:14,    Cycle = 1
-{% endhighlight %}
-
-**Finding TKey objects**
-
-With the `TFile::Get()` method, you can find [TKey](https://root.cern/doc/master/classTKey.html){:target="_blank"} objects.
-
-_**Example**_
-
-{% highlight C++ %}
-   root[] TH1F *h9 = (TH1F*)f.Get("h9");
-{% endhighlight %}
-
-The `Get()` method finds the `TKey` object with name **h9**.
-
-
-In detail, the following happens when executing `GetObject()`:
-
-- The key with name `hpx;1` is found in the list of keys.
-
-- A {% include ref class="TBuffer" %}object is created.
-
-- The buffer is read from the ROOT file.
-
-- An empty object is created by calling the default constructor for the class referenced in {% include ref class="TKey" %}.
-
-- The [Streamer()](https://root.cern/doc/master/classTClass.html#ac1c95f1787550ebc5367590aedacbd67){:target="_blank"} method is called for this new object.
-
-In case there is an object with multiple cycles, you can pick a particular cycle with a name like `hpx;` (for example `hpx;2`).
-
- You can also directly access the keys, for example when the names of the objects contained in the ROOT file are not known or when a long series of objects needs to be read sequentially.
-
- _**Example**_
-
-This example illustrates how to loop over all keys of a ROOT file.
-
-{% highlight C++ %}
-for (TObject* keyAsObj : *inputFile.GetListOfKeys()){
-    auto key = dynamic_cast<TKey*>(keyAsObj);
-    std::cout << "Key name: " << key->GetName() << " Type: " << key->GetClassName() << std::endl;
-}
+...
 {% endhighlight %}
 
-
-
-**Iterating over objects**
-
-Keys are available in a {% include ref class="TList" %} of {% include ref class="TKey" %}s. Therefore, you can iterate over the list of keys.
+For instance if you do not know an object's name upfront, it can be useful to iterate through all of a directory's entries.
 
 _**Example**_
 
-The `TKeys` of the `demo.root` file are iterated.
-
 {% highlight C++ %}
-{
-   TFile f("demo.root");
-   TIter next(f.GetListOfKeys());
-   TKey *key;
-   while ((key=(TKey*)next())) {
-      printf("key: %s points to an object of class: %s at %dn", key->GetName(),
-      key->GetClassName(),key->GetSeekKey());
-   }
+TFile f("demo.root");
+// f.GetListOfKeys() contains `TObject*`, but we know they are really `TKey*`
+for (TKey *key: ROOT::RangeStaticCast<TKey*>(*f.GetListOfKeys())) {
```

Comment:
```suggestion
for (auto key: ROOT::RangeStaticCast<TKey*>(*f.GetListOfKeys())) {
```

Reply:
we actually discussed this and decided to go with explicit types (e.g. here to make it clear that `RangeStaticCast<TKey*>` returns an iterator over `TKey *`, and not `TKey&` for instance